orttraining/orttraining/test/python/orttraining_test_orttrainer_frontend.py

```diff
@@ -589,85 +597,139 @@ def testInstantiateORTTrainer(step_fn, lr_scheduler, expected_lr_values):
     assert (onnx.helper.printable_graph(trainer_from_onnx._onnx_model.graph) == onnx.helper.printable_graph(trainer._onnx_model.graph))
 
 
-@pytest.mark.parametrize("seed, device_id", [
+@pytest.mark.parametrize("seed, device", [
     (0, 'cpu'),
-    (42, 'cpu'),
-    (0, 'cuda:0'),
     (24, 'cuda')
 ])
-def testORTDeterministicCompute(seed, device_id):
+def testORTDeterministicCompute(seed, device):
+    # Common setup
     optim_config = optim.LambConfig()
     opts = orttrainer.ORTTrainerOptions({
         'debug' : {
             'deterministic_compute': True
         },
         'device' : {
-            'id' : device_id,
+            'id' : device,
             'mem_limit' : 10*1024*1024
         }
     })
-    
+
+    # Setup for the first ORTTRainer run
     torch.manual_seed(seed)
     set_seed(seed)
-
-    # Using PyTorch Transformer model as example
-    model, model_desc, trainer, data, targets = generate_pytorch_transformer_model_sample(optim_config, opts,  device=device_id)
-
-    # Run first model train step
-    output = trainer.train_step(data, targets)
-    assert trainer._onnx_model is not None
+    model, model_desc, my_loss, batcher_fn, train_data, val_data, _ = _load_pytorch_transformer_model(device)
+    first_trainer = orttrainer.ORTTrainer(model, model_desc, optim_config, loss_fn=my_loss, options=opts)
+    data, targets = batcher_fn(train_data, 0)
+    _ = first_trainer.train_step(data, targets)
+    assert first_trainer._onnx_model is not None
     
-    # Reset the seeds
+    # Setup for the second ORTTRainer run
     torch.manual_seed(seed)
     set_seed(seed)
-
-    # Run second model train step
-    _, _, second_trainer, _, _ = generate_pytorch_transformer_model_sample(optim_config, opts, device=device_id)
-    output = second_trainer.train_step(data, targets)
+    model, _, _, _, _, _, _ = _load_pytorch_transformer_model(device)
+    second_trainer = orttrainer.ORTTrainer(model, model_desc, optim_config, loss_fn=my_loss, options=opts)
```

Comment:
Can we use the same instance of the PyTorch model? Or do a deepcopy instead?

Reply:
this is a pytorch model, which is not changed by ORT and will be used to generate an ONNX model. They can be the same as there is no risk of one run affect the other